### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by kryvets-vadym**
[DEMO LINK](https://kryvets-vadym.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -26,70 +28,127 @@ type ReorderOptions = {
...
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {orderedGoods.map(good => (
+              <li data-cy="Good">{ good }</li>
+            ))}
```
Comment (Stepico) [inline src/App.tsx:148]: I think the key attribute is must-have

File: src/App.tsx [hunk 2]
```diff
@@ -26,70 +28,127 @@ type ReorderOptions = {
...
+              'is-light': !isReversed,
+            })}
+            onClick={() => reverse()}
+          >
+            Reverse
+          </button>
+
+          {sortType === SortType.NONE && !isReversed
+            ? ('')
+            : (
```
Comment (Stepico) [inline src/App.tsx:133]: I don't think that you need ternary here because, use conditional rendering

**Review by Stepico — CHANGES_REQUESTED** [review]
Seems good to me, just some minor suggestions from me

**Review by Stepico — APPROVED** [review]
GJ, also I noticed that you can click on an active button and it becomes inactive and it works with Sort buttons, but the mockup doesn't show it, yet it's better that you added some additional functionality, but reset button in this way is useless I guess, so I'm not sure about this one

File: src/App.tsx [hunk 3]
```diff
@@ -26,70 +28,126 @@ type ReorderOptions = {
...
+  setSortType = (sortType: SortType) => {
+    return this.state.sortType === sortType
+      ? this.setState({
+        sortType: SortType.NONE,
+      })
+
+      : this.setState({
+        sortType,
+      });
+  };
```
Comment (mykola-koval) [inline src/App.tsx:84]: I guess we should simplify this or even create a function for each type of sorting

Comment (mykola-koval) [reply src/App.tsx:84]: right now i dont see any benefits in using this function over regular setState

File: src/App.tsx [hunk 4]
```diff
@@ -26,70 +28,126 @@ type ReorderOptions = {
...
+            onClick={() => reverse()}
+          >
+            Reverse
+          </button>
+
+          {isOrderChanged && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => reset()}
```
Comment (mykola-koval) [inline src/App.tsx:136]:
```suggestion
              onClick={reset}
```

File: src/App.tsx [hunk 5]
```diff
@@ -26,70 +28,126 @@ type ReorderOptions = {
...
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button', 'is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={() => reverse()}
```
Comment (mykola-koval) [inline src/App.tsx:127]:
```suggestion
            onClick={reverse}
```

File: src/App.tsx [hunk 6]
```diff
@@ -26,70 +28,126 @@ type ReorderOptions = {
...
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {orderedGoods.map(good => (
+              <li data-cy="Good" key={good}>{ good }</li>
```
Comment (mykola-koval) [inline src/App.tsx:146]:
```suggestion
              <li data-cy="Good" key={good}>
                {good}
              </li>
```

File: src/App.tsx [hunk 7]
```diff
@@ -26,70 +27,128 @@ type ReorderOptions = {
...
+  render() {
+    const { sortType, isReversed } = this.state;
+    const {
+      sortByAlphabet,
+      sortByLength,
+      reverse,
+      reset,
+    } = this;
+    const isOrderChanged = isReversed || sortType !== SortType.NONE;
+    const orderedGoods = getReorderedGoods(goodsFromServer, this.state);
```
Comment (denys-danyliuk) [inline src/App.tsx:98]:
```suggestion
    const { sortType, isReversed } = this.state;
    const {
      sortByAlphabet,
      sortByLength,
      reverse,
      reset,
    } = this;
    
    const isOrderChanged = isReversed || sortType !== SortType.NONE;
    
    const orderedGoods = getReorderedGoods(goodsFromServer, this.state);
```

File: src/App.tsx [hunk 8]
```diff
@@ -26,70 +27,128 @@ type ReorderOptions = {
...
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button', 'is-info', {
+              'is-light': sortType !== SortType.ALPHABET,
+            })}
```
Comment (denys-danyliuk) [inline src/App.tsx:107]:
```suggestion
            className={classNames('button', 'is-info', 
            {
              'is-light': sortType !== SortType.ALPHABET,
            }
          )}
```

File: src/App.tsx (hunk 6 above)
Comment (denys-danyliuk) [reply src/App.tsx:147]: fix it

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**